guarddog/analyzer/analyzer.py

```diff
@@ -32,14 +34,18 @@ class Analyzer:
 
     def __init__(self, ecosystem=ECOSYSTEM.PYPI) -> None:
         self.sourcecode_rules_path = os.path.join(os.path.dirname(__file__), "sourcecode")
-
         self.ecosystem = ecosystem
 
         # Rules and associated detectors
         self.metadata_detectors = get_metadata_detectors(ecosystem)
 
         self.metadata_ruleset: set[str] = set(self.metadata_detectors.keys())
-        self.sourcecode_ruleset: set[str] = set(rule["id"] for rule in SOURCECODE_RULES[ecosystem])
+        self.semgrep_ruleset: set[str] = set(
+            [r.id for r in get_sourcecode_rules(ecosystem, SempgrepRule)]
+        )
+        self.yara_ruleset: set[str] = set(
+            [r.id for r in get_sourcecode_rules(ecosystem, YaraRule)]
```

Comment:
I think you can just do `set(r.id for r in get_sourcode_rules(...))`

Reply:
done in [aecc17c](https://github.com/DataDog/guarddog/pull/401/commits/aecc17cd62cc8b49845e267dec21b50f37ab0c11)